packages/react-dom/src/client/ReactDOMFiberTextarea.js

```diff
@@ -116,28 +118,24 @@ export function initWrapperState(element: Element, props: Object) {
   }
 
   node._wrapperState = {
-    initialValue: '' + initialValue,
+    initialValue: getSafeValue(initialValue),
   };
 }
 
 export function updateWrapper(element: Element, props: Object) {
   const node = ((element: any): TextAreaWithWrapperState);
-  const value = props.value;
+  const value = getSafeValue(props.value);
   if (value != null) {
-    // Cast `value` to a string to ensure the value is set correctly. While
-    // browsers typically do this as necessary, jsdom doesn't.
-    const newValue = '' + value;
-
     // To avoid side effects (such as losing text selection), only set value if changed
-    if (newValue !== node.value) {
-      node.value = newValue;
+    if ((value: any) !== node.value) {
```

Comment:
We compared this against a `string` before and now it's a `SafeValue`. Is this intended? I think this means when value is a non-string (i.e a number) it will always update `node.value` since the strict equality will always return false.

Reply:
Hm, right. Would it make more sense if the comparison was `props.value !== node.value` and use `getSafeValue` to for `const newValue`?

Reply:
That would still false if `props.value = 1` and `node.value = '1'`. I think bringing back `const newValue` as `safeValueToString(value)` is more correct.